lib/database_consistency/checkers/enum_checkers/enum_column_checker.rb

```diff
@@ -0,0 +1,49 @@
+# frozen_string_literal: true
+
+module DatabaseConsistency
+  module Checkers
+    # This class checks that ActiveRecord enum is backed by enum column
+    class EnumColumnChecker < EnumChecker
+      Report = ReportBuilder.define(
+        DatabaseConsistency::Report,
+        :table_name,
+        :column_name
+      )
+
+      private
+
+      # ActiveRecord supports native enum type since version 7 and only for PostgreSQL
+      def preconditions
+        Helper.postgresql? && ActiveRecord::VERSION::MAJOR >= 7 && column.present?
```

Comment:
I remember we discussed enabling this checker only for PostgreSQL 15+, but let me do some benchmarks for earlier versions as well. I think we should stick to "postgres and AR >=7" condition here.

Reply:
> I remember we discussed enabling this checker only for PostgreSQL 15+, but let me do some benchmarks for earlier versions as well.

This would be awesome!

Reply:
> I think we should stick to "postgres and AR >=7" condition here.

Yes, we can do that 👍